### PR TITLE
Quad4 element and simple example

### PR DESCRIPTION
**Pull request by pmetsis**
Good morning everyone! I am uploading the Quad4 plane stress element. I also had to make some changes to quadrature and material. You can check the element by solving the simple example provided. Since I use Xamarin for Mac (and I haven't figured out the git extension for it yet), I have manually uploaded these files, if something else is missing please notify. Stresses/strains are not yet implemented but I couldn't comment them out for consistency reasons. I will also upload the solution of the simple example by hand or using some Matlab routines of my own in order to verify the code (I have already checked that it works).

### CODE REVIEW

File: ElasticMaterial2D.cs [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        private readonly double[] stresses = new double[3];
+        private double[,] constitutiveMatrix = null;
+        public double YoungModulus { get; set; }
+        public double PoissonRatio { get; set; }
+        public double[] Coordinates { get; set; }
+
+        private double[,] GetConstitutiveMatrix()
+        {
+
+            //Panos Plane Stress
```
Comment (georgestavroulakis) [inline ElasticMaterial2D.cs:23]: Comments like that should be left out. The proper place for such stuff is the documentation which will be constructed

File: ElasticMaterial2D.cs [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+            // [D] = E/(1-v^2)    [ v       1       0   ]
+            //                    [ 0       0   (1-v)/2 ]
+            //
+
+            double fE1 = YoungModulus / (1 - PoissonRatio*PoissonRatio);
+            double fE2 = (1 - PoissonRatio)/2;
+            double[,] afE = new double[3, 3];
+            afE[0, 0] = fE1;
+            afE[0, 1] = fE1 * PoissonRatio;
+            //afE[0, 2] = 0;
```
Comment (georgestavroulakis) [inline ElasticMaterial2D.cs:35]: Please remove code that is commented out.

File: GaussQuadrature.cs [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+        public double Zeta { get; private set; }
+        public double WeightFactor { get; private set; }
+        public double[,] DeformationMatrix { get; private set; }
+        // 
+        #endregion
+    }
+
+    public class GaussQuadrature
+    {
+        /* For point coordinates, we encounter the following constants:
```
Comment (georgestavroulakis) [inline GaussQuadrature.cs:72]: These comments should be added to the documentation

File: Quad4.cs [hunk 4]
```diff
@@ -0,0 +1,619 @@
...
+using ISAAR.MSolve.Matrices;
+using ISAAR.MSolve.PreProcessor.Elements.SupportiveClasses;
+
+namespace ISAAR.MSolve.PreProcessor.Elements
+{
+	public class Quad4 : IStructuralFiniteElement, IEmbeddedHostElement
+	{
+		protected static double determinantTolerance = 0.00000001;
+
+		protected static int iInt = 2;
```
Comment (georgestavroulakis) [inline Quad4.cs:18]: This should be changed and input to the constructor (maybe have multiple constructors with one of them having the default gauss integration order)

File: Quad4.cs [hunk 5]
```diff
@@ -0,0 +1,619 @@
...
+            : this(material)
+        {
+			this.dofEnumerator = dofEnumerator;
+		}
+        public IFiniteElementDOFEnumerator DOFEnumerator
+		{
+			get { return dofEnumerator; }
+			set { dofEnumerator = value; }
+		}
+        //Panos - I dont think I need these
```
Comment (georgestavroulakis) [inline Quad4.cs:48]: These properties should be put to an interface that will need to be implemented for elements that support dynamic analysis

File: Quad4.cs [hunk 6]
```diff
@@ -0,0 +1,619 @@
...
+				faXY[0, i] = element.Nodes[i].X;
+				faXY[1, i] = element.Nodes[i].Y;
+			}
+			return faXY;
+		}
+
+		#region IElementType Members
+
+		public int ID
+		{
```
Comment (georgestavroulakis) [inline Quad4.cs:78]: All ID fields should be deleted

File: Quad4.cs [hunk 7]
```diff
@@ -0,0 +1,619 @@
...
+			return dofTypes;
+		}
+        public IList<Node> GetNodesForMatrixAssembly(Element element)
+		{
+			return element.Nodes;
+		}
+            
+        private double[] CalcQ4Shape(double fXi, double fEta)
+        {
```
Comment (georgestavroulakis) [inline Quad4.cs:98]: This should be put to the documentation

File: Quad4.cs [hunk 8]
```diff
@@ -0,0 +1,619 @@
...
+				{
+					coordinateDifferences[0] += shapeFunctions[i] * element.Nodes[i].X;
+					coordinateDifferences[1] += shapeFunctions[i] * element.Nodes[i].Y;
+				}
+				coordinateDifferences[0] = node.X - coordinateDifferences[0];
+				coordinateDifferences[1] = node.Y - coordinateDifferences[1];
+
+				double[,] faXY = GetCoordinatesTranspose(element);
+				double[] ShapeFunctionDerivatives = CalcQ4ShapeFunctionDerivatives(naturalCoordinates[0], naturalCoordinates[1]);
+                //SOS PANOS - THIS IS WRONG, I SHOULD IMPLEMENT INVERSE JACOBIAN
```
Comment (georgestavroulakis) [inline Quad4.cs:597]: We should implement a generic inverse jacobian static class

**Review by GoatOfRafin — CHANGES_REQUESTED** [review]
Minor changes need to be implemented.

**Comment by SerafeimBakalakos** [comment]
This pull request has been stagnant for 2 years and the extra features have been added by other pull requests. Therefore I am closing this.

**Pull request closed without merging**